src/search/Search.tsx

```diff
@@ -99,30 +97,30 @@ export function Search({ controller, onClose }: SearchProps) {
             onClick={(event) => {
                 const target = event.target as HTMLElement;
                 if (target.id == "search-dialog") { 
-                    _onClose() 
+                    onCloseRequest()
                 }
                 const LI = target.closest("li");
                 if (LI != null) {
                     const index = LI.getAttribute("data-index");
                     if (index) {
                         results[parseInt(index)].action();
-                        setTimeout(_onClose, 1);
+                        setTimeout(onCloseRequest, 1);
                     }
                 }
             }}
-            className={`fixed z-50 inset-0 bg-gray-900 bg-opacity-60 overflow-y-auto h-full w-full px-4 ${visible ? '' : 'hidden'}`}>
+            className={`fixed z-50 inset-0 bg-gray-900 bg-opacity-60 overflow-y-auto h-full w-full px-4`}>
             <div className="relative top-20 mx-auto shadow-xl rounded-md bg-zinc-700 max-w-md p-1 flex flex-col">
-                {controller.prompt.length > 0 ?
+                {prompt.length > 0 ?
                     <h1 className="p-1 text-lg py-2">
-                        {controller.prompt}
+                        {prompt}
                     </h1> : null
                 }
                 <input
                     type="text"
                     value={state.query}
                     onChange={(event) => {
                         const value = event.target.value;
-                        _onSearch(value);
+                        onSearch(value);
                         updateState({ ...state, query: value });
```

Comment:
można to uprościć do updateState({ ...state, query: event.target.value }), zapiąć useEffect na state.value  i wywoływać  w nim onSearch(state.value)

Reply:
Nawet działa, fajny trik, dzieki :)